packages/pagination/src/usePagination.ts

```diff
@@ -5,17 +5,45 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import { useSelection } from '@zendeskgarden/container-selection';
+import {
+  useSelection,
+  IUseSelectionProps,
+  IUseSelectionState,
+  IGetItemPropsOptions
+} from '@zendeskgarden/container-selection';
 
-export function usePagination(options) {
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface IUsePaginationProps<Item> extends IUseSelectionProps<Item> {}
+
+export interface IGetPageProps<Item> extends IGetItemPropsOptions<Item> {
+  page?: number;
+  current?: any;
+  ariaLabel?: any;
+}
+
+export interface IGetContainerProps extends React.HTMLProps<any> {
+  role?: any;
+  ariaLabel?: string;
+}
+
+export interface IUsePaginationReturnValue<Item> extends IUseSelectionState<Item> {
+  getContainerProps: <T>(options?: T & IGetContainerProps) => any;
```

Comment:
Is there a reason that this doesn't follow the `T extends IGetContainerProps` syntax shown in the other prop-getters?

Reply:
It turns out that `getContainerProps` does not extend from `IGetContainerProps` because unlike, `getPreviousPageProps`, `getNextPageProps`, `getPageProps`, it does not have any required attributes so TypeScript would complain when an unknown attribute such as `data-garden-id` was added.

@austin94 and I paired together and he suggested that what I have works, but cleaned the code a little bit by removing the generic `<T>`:

```tsx
getContainerProps: (options?: any & IGetContainerProps) => any;
```